pkg/cloud/aliyunprovider.go

```diff
@@ -0,0 +1,828 @@
+	}
+
+	var clusterName string
+	if c.ClusterName != "" {
+		clusterName = c.ClusterName
+	}
+
+	// Use a default name if none has been set until this point
+	if clusterName == "" {
+		clusterName = "Aliyun Cluster #1"
+	}
+
+	m := make(map[string]string)
+	m["name"] = clusterName
+	m["provider"] = kubecost.AliyunProvider
+	m["project"] = aliyun.clusterAccountId
+	m["region"] = aliyun.clusterRegion
+	m["id"] = env.GetClusterID()
+	return m, nil
+}
+
+// Will look at this in Next PR if needed
+func (aliyun *Aliyun) GetAddresses() ([]byte, error) {
+	return nil, nil
+}
+
+// Will look at this in Next PR if needed
+func (aliyun *Aliyun) GetDisks() ([]byte, error) {
+	return nil, nil
+}
+
+// Will look at this in Next PR if needed
+func (aliyun *Aliyun) UpdateConfig(r io.Reader, updateType string) (*CustomPricing, error) {
+	return nil, nil
+}
+
+// Will look at this in Next PR if needed
+func (aliyun *Aliyun) UpdateConfigFromConfigMap(cm map[string]string) (*CustomPricing, error) {
+	return nil, nil
+}
+
+// Will look at this in Next PR if needed
+func (aliyun *Aliyun) GetManagementPlatform() (string, error) {
+	return "", nil
+}
+
+// Will look at this in Next PR if needed
+func (aliyun *Aliyun) GetLocalStorageQuery(window, offset time.Duration, rate bool, used bool) string {
+	return ""
+}
+
+// Will look at this in Next PR if needed
+func (aliyun *Aliyun) ApplyReservedInstancePricing(nodes map[string]*Node) {
+
+}
+
+// Will look at this in Next PR if needed
+func (aliyun *Aliyun) ServiceAccountStatus() *ServiceAccountStatus {
+	return &ServiceAccountStatus{}
+}
+
+// Will look at this in Next PR if needed
+func (aliyun *Aliyun) PricingSourceStatus() map[string]*PricingSource {
+	return map[string]*PricingSource{}
+}
+
+// Will look at this in Next PR if needed
+func (aliyun *Aliyun) ClusterManagementPricing() (string, float64, error) {
+	return "", 0.0, nil
+}
+
+// Will look at this in Next PR if needed
+func (aliyun *Aliyun) CombinedDiscountForNode(string, bool, float64, float64) float64 {
+	return 0.0
+}
+
+type AliyunNodeKey struct {
+	ProviderID       string
+	RegionID         string
+	InstanceType     string
+	OSType           string
+	OptimizedKeyword string //If IsIoOptimized key will have optimize if not unoptimized the key for the node
+}
+
+func NewAliyunNodeKey(node *SlimK8sNode, optimizedKeyword string) *AliyunNodeKey {
+	return &AliyunNodeKey{
+		ProviderID:       node.ProviderID,
+		RegionID:         node.RegionID,
+		InstanceType:     node.InstanceType,
+		OSType:           node.OSType,
+		OptimizedKeyword: optimizedKeyword,
+	}
+}
+
+func (aliyunNodeKey *AliyunNodeKey) ID() string {
+	return aliyunNodeKey.ProviderID
+}
+
+func (aliyunNodeKey *AliyunNodeKey) Features() string {
+	keyLookup := stringutil.DeleteEmptyStringsFromArray([]string{aliyunNodeKey.RegionID, aliyunNodeKey.InstanceType, aliyunNodeKey.OSType, aliyunNodeKey.OptimizedKeyword})
+	return strings.Join(keyLookup, "::")
```

Comment:
Does Ali support some of these parameters (e.g. InstanceType) being absent from this "lookup" key? Is that why we have the `DeleteEmptyStringsFromArray` here?

Reply:
It was just to be extra caution, if node data is not populated with information but this function can be removed!

Reply:
Whatever you feel is safer works for me, just want to make sure we had thought about it.